src/stores/AllEventsStore.test.js

```diff
@@ -0,0 +1,91 @@
+import { Routes } from 'constants';
+jest.mock('../network/graphql/queries'); // block and manually mock our backend
+import AllEventsStore from './AllEventsStore'; // eslint-disable-line
+
+/** mock necessary http needed modules */
+const localStorageMock = {
+  getItem: jest.fn(),
+  setItem: jest.fn(),
+  clear: jest.fn(),
+};
+const navigatorMock = {};
+global.navigator = navigatorMock;
+global.localStorage = localStorageMock;
+
+/** test */
+describe('AllEventsStore', () => {
+  let store;
+  const app = {
+    sortBy: 'ASC',
+    wallet: { addresses: 0 },
+    global: { syncBlockNum: 0 },
+    refreshing: false,
+    ui: { location: 0 },
+  }; // mock the appstore
+  beforeEach(() => {
+    store = new AllEventsStore(app); // create a new instance before each test case
+  });
+
+  /** all following test cases target specifically to the mock backend, eg. network/graphql/__mocks__/queries.js */
+  it('Init', async () => {
+    const curListNum = store.list.length;
+    expect(curListNum).toBe(0);
+    expect.assertions(4);
```

Comment:
curious, what does this do?

Reply:
Line 31 & 32 is trying to make sure the list is empty before init.
Line 33 says this test case should pass 4 assertions. This line is unnecessary for this test case but good to have. But in general, it's useful in asynchronous tests.
For example,
`it('calls the API and throws an error', async () => {
  expect.assertions(2);
  try {
    await login('email', 'password');
  } catch (error) {
    expect(error.name).toEqual('Unauthorized');
    expect(error.status).toEqual(401);
  }
});`
Without it, this test case will pass even when login successful.

Or 
`test('doAsync calls both callbacks', () => {
  expect.assertions(2);
  function callback1(data) {
    expect(data).toBeTruthy();
  }
  function callback2(data) {
    expect(data).toBeTruthy();
  }

  doAsync(callback1, callback2);
});`
Here it ensures that both callbacks are actually called.

Reply:
gotcha, its just saying how many asserts are required.